### PR TITLE
Grab bag of (primarily race) test fixes

### PR DESCRIPTION
**Pull request by notnoop**
Fixes to various tests, mainly data race related - better to be reviewed commit-by-commit.

These reveal no actual data race in code so grouping them here independently from others.

### CODE REVIEW

File: helper/escapingio/reader_test.go [hunk 1]
```diff
@@ -426,14 +426,16 @@ func naiveEscapeCharacters(input string, escapeChar byte, h Handler) string {
 	}
 
 	return reg.ReplaceAllStringFunc(input, func(match string) string {
-		if len(match) != 3 {
-			panic(fmt.Errorf("match isn't 3 characters: %s", match))
+		// match can be more than three bytes because of unicode
```
Comment (endocrimes) [inline helper/escapingio/reader_test.go]: if the intention is to support unicode, could we switch this to `utf8.RuneCountInString`?

File: client/fs_endpoint_test.go [hunk 2]
```diff
@@ -597,11 +601,14 @@ OUTER:
...
 type ReadWriteCloseChecker struct {
 	io.ReadWriteCloser
+	l      sync.Mutex
 	Closed bool
 }
 
 func (r *ReadWriteCloseChecker) Close() error {
+	r.l.Lock()
 	r.Closed = true
+	r.l.Lock()
```
Comment (endocrimes) [inline client/fs_endpoint_test.go]: `r.l.Unlock()`?

File: helper/escapingio/reader_test.go (hunk 1 above)
Comment (notnoop) [reply helper/escapingio/reader_test.go]: Nope - we meant to only process ascii characters for escaping purposes.

File: client/fs_endpoint_test.go (hunk 2 above)
Comment (notnoop) [reply client/fs_endpoint_test.go]: oops - i thought i fixed this :( - i'll repush.

**Pull request merged**